### PR TITLE
Instance provider refactoring

### PR DESCRIPTION
**Pull request by amandeeprehal-lab49**
Instance provider refactoring

### CODE REVIEW

File: e2e-test/business/FTDAO.ts [hunk 1]
```diff
@@ -169,15 +175,16 @@ export default class FTDAO extends BaseDao {
     contractToUpgrade: string,
     client: Client = clientsInfo.operatorClient,
     description: string = DEFAULT_DESCRIPTION,
-    link: string = DEFAULT_LINK
+    link: string = DEFAULT_LINK,
+    nftTokenSerialId: number
```
Comment (shivamgarglab49) [inline e2e-test/business/FTDAO.ts:179]: we can put default value for 'nftTokenSerialId' like we did above.

File: e2e-test/business/factories/FTTokenHolderFactory.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  protected getPrefix() {
+    return "GOD";
+  }
+
+  protected getHolderInstance(contractId: ContractId) {
+    return new GodHolder(contractId);
+  }
+
+  protected getHolderLogic() {
+    return ContractService.NFT_HOLDER;
```
Comment (shivamgarglab49) [inline e2e-test/business/factories/FTTokenHolderFactory.ts:21]: it should be 'GOD_HOLDER'

File: e2e-test/business/Base.ts [hunk 3]
```diff
@@ -11,22 +11,48 @@ import {
...
+  protected csDev: ContractService = new ContractService();
   protected htsAddress: string;
   protected configuration: string;
   contractId: string;
+  contractName: string;
   private UPGRADE_HEDERA_SERVICE = "upgradeHederaService";
   private OWNER = "owner";
 
-  constructor(_contractId: string) {
+  constructor(_contractId: ContractId | null = null) {
```
Comment (shivamgarglab49) [inline e2e-test/business/Base.ts:27]: it should also accept _contractId in string format

File: e2e-test/business/TextDao.ts [hunk 4]
```diff
@@ -13,6 +13,7 @@ import {
   ContractExecuteTransaction,
 } from "@hashgraph/sdk";
 import { Helper } from "../../utils/Helper";
+import { ContractService } from "../../deployment/service/ContractService";
```
Comment (shivamgarglab49) [inline e2e-test/business/TextDao.ts:17]: this whole file is not required now, right ?

File: e2e-test/business/ContractUpgradeDao.ts [hunk 5]
```diff
@@ -13,6 +13,7 @@ import {
   ContractExecuteTransaction,
```
Comment (shivamgarglab49) [inline e2e-test/business/ContractUpgradeDao.ts:13]: this whole file also not reqruied ?

File: e2e-test/business/FTDAO.ts (hunk 1 above)
Comment (amandeeprehal-lab49) [reply e2e-test/business/FTDAO.ts]: Done

File: e2e-test/business/factories/FTTokenHolderFactory.ts (hunk 2 above)
Comment (amandeeprehal-lab49) [reply e2e-test/business/factories/FTTokenHolderFactory.ts]: Done

File: e2e-test/business/Base.ts (hunk 3 above)
Comment (amandeeprehal-lab49) [reply e2e-test/business/Base.ts:27]: As discussed, we will use strong type here to enforce the contract.

File: e2e-test/business/TextDao.ts (hunk 4 above)
Comment (amandeeprehal-lab49) [reply e2e-test/business/TextDao.ts]: yes, deleted.

File: e2e-test/business/ContractUpgradeDao.ts (hunk 5 above)
Comment (amandeeprehal-lab49) [reply e2e-test/business/ContractUpgradeDao.ts]: Deleted, you are right!

**Review by shivamgarglab49 — APPROVED** [review]

**Pull request merged**